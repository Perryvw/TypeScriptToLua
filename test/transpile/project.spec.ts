import { resolveFixture, transpileProjectResult } from "./run";

test("should transpile", () => {
    const { diagnostics, emittedFiles } = transpileProjectResult(resolveFixture("project/tsconfig.json"));
    expect(diagnostics).not.toHaveDiagnostics();
    expect(emittedFiles).toMatchSnapshot();
});
