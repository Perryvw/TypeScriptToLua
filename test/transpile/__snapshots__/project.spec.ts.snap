// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should transpile 1`] = `
Array [
  Object {
    "name": "otherFile.lua",
    "text": "--[[ Generated with https://github.com/TypeScriptToLua/TypeScriptToLua ]]
local ____exports = {}
function ____exports.getNumber(self)
    return GetAPIValue()
end
return ____exports
",
  },
  Object {
    "name": "index.lua",
    "text": "--[[ Generated with https://github.com/TypeScriptToLua/TypeScriptToLua ]]
local ____exports = {}
local ____otherFile = require(\\"otherFile\\")
local getNumber = ____otherFile.getNumber
local myNumber = getNumber(nil)
SetAPIValue(myNumber * 5)
return ____exports
",
  },
]
`;
